priam/src/test/java/com/netflix/priam/backup/TestS3FileSystem.java

```diff
@@ -41,79 +47,102 @@
-    private static String region;
-
-    public TestS3FileSystem() {
-        if (injector == null) injector = Guice.createInjector(new BRTestModule());
+    @Inject private BackupMetrics backupMetrics;
+    @Inject private InstanceInfo instanceInfo;
+    @Inject private S3FileSystem s3FileSystem;
+    @Inject private S3BackupPath s3backupPath;
+    @Inject private IConfiguration configuration;
+    @ReplaceWithMock private AmazonS3Client amazonS3Client;
+    @ReplaceWithMock private S3PartUploader s3PartUploader;
+    private final String FILE_PATH =
+            "target/cass/data/Keyspace1/Standard1/backups/201108082320/Keyspace1-Standard1-ia-1-Data.db";
 
-        if (backupMetrics == null) backupMetrics = injector.getInstance(BackupMetrics.class);
-        InstanceInfo instanceInfo = injector.getInstance(InstanceInfo.class);
-        region = instanceInfo.getRegion();
-    }
+    private static String region;
 
-    @BeforeClass
-    public static void setup() throws InterruptedException, IOException {
+    @Before
+    public void setup() throws InterruptedException, IOException {
         new MockS3PartUploader();
         new MockAmazonS3Client();
 
-        File dir1 = new File("target/data/Keyspace1/Standard1/backups/201108082320");
-        if (!dir1.exists()) dir1.mkdirs();
         File file = new File(FILE_PATH);
+        file.getParentFile().mkdirs();
         long fiveKB = (5L * 1024);
         byte b = 8;
         BufferedOutputStream bos1 = new BufferedOutputStream(new FileOutputStream(file));
         for (long i = 0; i < fiveKB; i++) {
             bos1.write(b);
         }
         bos1.close();
+        region = instanceInfo.getRegion();
     }
 
-    @AfterClass
-    public static void cleanup() {
+    @After
+    public void cleanup() {
         File file = new File(FILE_PATH);
         file.delete();
     }
 
     @Test
-    public void testFileUpload() throws Exception {
-        MockS3PartUploader.setup();
-        IBackupFileSystem fs = injector.getInstance(NullBackupFileSystem.class);
-        S3BackupPath backupfile = injector.getInstance(S3BackupPath.class);
-        backupfile.parseLocal(new File(FILE_PATH), BackupFileType.SNAP);
-        long noOfFilesUploaded = backupMetrics.getUploadRate().count();
-        fs.uploadFile(
-                Paths.get(backupfile.getBackupFile().getAbsolutePath()),
-                Paths.get(backupfile.getRemotePath()),
-                backupfile,
+    public void testS3BackupPathParser() throws Exception {
+        s3backupPath.parseLocal(new File(FILE_PATH), BackupFileType.SNAP);
+        Assert.assertEquals("Keyspace1", s3backupPath.keyspace);
+        Assert.assertEquals("Standard1", s3backupPath.columnFamily);
+        Assert.assertNotNull(s3backupPath.getTime());
+    }
+
+    @Test
+    public void testFilePutUploadSuccess() throws Exception {
+        double success = backupMetrics.getValidUploads().actualCount();
+        MockAmazonS3Client.putFailure = false;
```

Comment:
`MockIto` might be more easy and comfortable here.

        Mockito.when(mockAmazonS3Client. putFailure()).thenReturn(false);